### PR TITLE
Add failure history to template

### PR DESCRIPTION
**Pull request by jkakavas**
Debugging a CI issue, it's very often helpful to know when a
failure started happening and how often it happens. As such it is
beneficial to add a relevant section in the template in order to
remind the person on test triage to share a link to the build
stats Kibana instance.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/test-failure.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ to reproduce locally
 
 **Applicable branches**:
 
+**Failure history**:
+Link to build stats and possible indication of when this started failing and how often it fails
```
Comment (rjernst) [inline .github/ISSUE_TEMPLATE/test-failure.md:22]: I think this should be in a comment block like the "fill out" instructions above.

Comment (jkakavas) [reply .github/ISSUE_TEMPLATE/test-failure.md:23]: addressed in ac71c20

**Comment by jkakavas** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**